app/repositories/repository_challenge.rb

```diff
@@ -0,0 +1,69 @@
+module RepositoryChallenge
+
+  def self.collection_aggregate(*args)
+    Challenge.collection.aggregate(args.flatten)
+  end
+
+  def self.count_entries_query
+    collection_aggregate({
+      "$group" => {
+        "_id" => nil,
+        "count_entries" => {
+          "$sum" => {
+            "$size" =>  { "$ifNull": [ "$entries", [] ] }
+          },
+        },
+      },
+    })
+  end
+
+  def self.count_entries
+    result = count_entries_query.first
+
+    result ? result['count_entries'] : 0
+  end
+
+  def self.score_query
```

Comment:
Nifty query! Ditto, could you provide a short docs explainer of what this does? For future reference and folks that are not familiar with Mongo..

Reply:
Yes

Reply:
done